### PR TITLE
Fixed an issue about the component(RangePicker) could not select the …

### PR DESCRIPTION
**Pull request by alex89lj**
…same month

First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch.
Pull request will be merged after one of collaborators approve.
Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

1.https://github.com/react-component/calendar/pull/478
2.https://github.com/ant-design/ant-design/issues/13302
3.https://github.com/ant-design/ant-design/pull/15398

 I added a submit for feature branch

### 📝 Changelog description

> Describe changes from userside, and list all potential break changes or other risks.

1. English description
Fixed an issue about the component(RangePicker)  could not select the same month
2. Chinese description (optional)

### ☑️ Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by zombieJ** [comment]
Test failed. Please check.

**Comment by alex89lj** [comment]
Well, I'll fix it

**Comment by alex89lj** [comment]
恩...  看了下 确实是的  本来是改月份模式的  却又影响到了日期模式 而且琢磨了下日期模式面板的设计又确实应该如此 RangePicker组件改起来几乎是一环套一环的 确实挺多细节需要注意 我刚改了下 本地测试已经通过了之前的快照 我再看下了 稍后提交

File: components/date-picker/__tests__/RangePicker.test.js [hunk 1]
```diff
@@ -319,4 +319,21 @@ describe('RangePicker', () => {
     const wrapper = mount(<RangePicker separator="test" />);
     expect(wrapper.render()).toMatchSnapshot();
   });
+
+  // https://github.com/ant-design/ant-design/issues/13302
+  it('in "month" mode，left panel and right panel should be the same month', () => {
```
Comment (afc163) [inline components/date-picker/__tests__/RangePicker.test.js]: `should be` => `could be`

File: components/date-picker/RangePicker.tsx [hunk 2]
```diff
@@ -21,14 +21,18 @@ export interface RangePickerState {
...
   // value could be an empty array, then we should not reset showDate
   if (!start && !end) {
     return;
   }
   const newEnd = end && end.isSame(start, 'month') ? end.clone().add(1, 'month') : end;
-  return [start, newEnd] as RangePickerValue;
+  if (mode && mode[0] === 'month') {
+    return [start, end] as RangePickerValue;
+  } else {
+    return [start, newEnd] as RangePickerValue;
```
Comment (zombieJ) [inline components/date-picker/RangePicker.tsx]: newEnd 可以移到下面来，减少计算。

File: components/date-picker/__tests__/RangePicker.test.js [hunk 3]
```diff
@@ -319,4 +319,21 @@ describe('RangePicker', () => {
...
+        .find('.ant-calendar-range-picker-input')
+        .first()
+        .getDOMNode().value,
+    ).toBe(
+      wrapper
+        .find('.ant-calendar-range-picker-input')
+        .last()
+        .getDOMNode().value,
+    );
+  });
```
Comment (zombieJ) [inline components/date-picker/__tests__/RangePicker.test.js]: 弹出来的面板最好也检测一下两边是不是都可以点。

Comment (alex89lj) [reply components/date-picker/__tests__/RangePicker.test.js]: 好的  我晚点添加下

Comment (zombieJ) [reply components/date-picker/__tests__/RangePicker.test.js]: 这两个 it 用一个 describe 包一起哈~

Comment (alex89lj) [reply components/date-picker/__tests__/RangePicker.test.js]: 好的哈  我稍后添加下

**Review by zombieJ — APPROVED** [review]

**Pull request merged**